extensions/amp-video/0.1/flexible-bitrate.js

```diff
@@ -213,6 +219,26 @@ export class BitrateManager {
       dev().fine(TAG, 'Playing at lower bitrate %s', video.currentSrc);
     });
   }
+
+  /**
+   * Update other managed videos when we learn that the current selected
+   * bandwidth wasn't good. Only operates on videos that are currently paused
+   * as we never want to interrupt playing videos if we don't have to.
+   */
+  updateOtherManagedAndPausedVideos_() {
+    this.videos_.forEach((weakref, index) => {
```

Comment:
Doing a forEach while mutating will cause index skips. This needs to be a manual for-loop with a `i--` to guarantee every element is checked.

```js
a = [1, 2, 3];
a.forEach((e, i) => {
    if (e >= 2) a.splice(i, 1);
});
assert.deepEqual(a, [1, 3]);
```

Reply:
Ha, I vaguely recalled this, but my too naive testing failed to detect this :)